lib/gc.cpp

```diff
@@ -56,10 +56,17 @@ extern "C" size_t GC_get_heap_size(void);
 extern "C" int GC_print_stats;
 
 static void gc_callback() {
-    size_t count;
-    LOG1("****** GC called ****** (heap size " << GC_get_heap_size() << ")");
-    LOG2("cstring cache size " << cstring::cache_size(count) << " (count=" << count << ")");
-    GC_print_stats = LOGGING(2) ? 1 : 0;  // unfortunately goes directly to stderr!
+    if (Log::verbose())
```

Comment:
To truly do it safely I think we'll need to use C APIs and get rid of iostreams in this function, because I doubt that there's any guarantee that iostreams doesn't allocate.